core/src/main/java/org/apache/iceberg/jdbc/JdbcClientPool.java

```diff
@@ -43,8 +53,16 @@ public JdbcClientPool(String dbUrl, Map<String, String> props) {
   }
 
   public JdbcClientPool(int poolSize, String dbUrl, Map<String, String> props) {
-    super(poolSize, SQLNonTransientConnectionException.class, true);
+    super(poolSize, SQLTransientException.class, true);
     properties = props;
+    retryableStatusCodes = Sets.newHashSet();
+    retryableStatusCodes.addAll(COMMON_RETRYABLE_CONNECTION_SQL_STATES);
+    String configuredRetryableStatuses = props.get(JdbcCatalog.RETRYABLE_STATUS_CODES);
+    if (configuredRetryableStatuses != null) {
```

Comment:
I'm presenting this configuration option since at the moment I believe it's the most practical way for people to work around production issues they hit today.

Long term, if we separate JDBC catalog into a separate library (I forgot where we landed on where JDBC fits in in terms of the catalog discussion that's happened in the past) that library could define status codes for each database so a user doesn't have to specify anything.